crates/core/src/kernel/snapshot/mod.rs

```diff
@@ -337,6 +339,20 @@ impl Snapshot {
     }
 }
 
+/// TODO!
+pub trait ReplayVistor: Send {
+    /// TODO!
+    fn visit_batch(&mut self, batch: &RecordBatch) -> DeltaResult<()>;
+}
+
+struct PrintVistor {}
+impl ReplayVistor for PrintVistor {
+    fn visit_batch(&mut self, _batch: &RecordBatch) -> DeltaResult<()> {
+        println!("Hello world!");
```

Comment:
:question:

Reply:
It's still WIP / draft. Opened this get some feedback on the new mechanism of implementing a visitor pattern for hooking into the log replay.